compiler/rustc_errors/src/lib.rs

```diff
@@ -1898,6 +1898,9 @@ pub fn report_ambiguity_error<'a, G: EmissionGuarantee>(
     for help_msg in ambiguity.b1_help_msgs {
         db.help(help_msg);
     }
+    if ambiguity.extern_crate {
+        return;
```

Comment:
Why is the rest of the function skipped?
What happens if it is not skipped?

Reply:
The ambiguity information is missing for cross-crates. It stores which binding is ambiguous but doesn't store the other ambiguous binding, so it's skipped.

Reply:
> What happens if it is not skipped?

Print the redundant message, because the `﻿ambiguity` field is pointing to itself.